jopa-impl/src/main/java/cz/cvut/kbss/jopa/model/metamodel/AnnotatedAccessor.java

```diff
@@ -0,0 +1,112 @@
+package cz.cvut.kbss.jopa.model.metamodel;
+
+import cz.cvut.kbss.jopa.exception.MetamodelInitializationException;
+
+import java.beans.Introspector;
+import java.lang.reflect.Method;
+
+/**
+ * Class used for extracting field name from getter or setter.
+ */
+
+public abstract class AnnotatedAccessor {
+    protected final Method method;
+    protected final String propertyName;
+    protected final Class<?> propertyType;
+
+    protected AnnotatedAccessor(Method method) {
+        this.method = method;
+        this.propertyName = extractPropertyNameFromMethod(method);
+        this.propertyType = extractPropertyTypeNameFromMethod(method);
+    }
+
+    public static AnnotatedAccessor from(Method method) {
+        if (isSetter(method)) {
+            return new AnnotatedSetter(method);
+        } else if (isGetter(method)) {
+            return new AnnotatedGetter(method);
+        } else {
+            throw new MetamodelInitializationException("Method {} is neither a setter or getter. Only simple getters and setters following the Java naming convection can be used for OWL property annotations.");
+        }
+
+    }
+
+    protected abstract String extractPropertyNameFromMethod(Method m);
+
+    protected abstract Class<?> extractPropertyTypeNameFromMethod(Method m);
+
+    private static boolean isSetter(Method m) {
+        return m.getName().startsWith("set") && m.getParameterCount() == 1;
+    }
+
+    private static boolean isGetter(Method m) {
+        if (m.getParameterCount() != 0 || m.getReturnType().equals(Void.TYPE)) { /// getter has 0 arguments and returns something
+            return false;
+        }
+
+        if (m.getName().startsWith("get")) {
```

Comment:
Use constants for these string literals. I see them reused in this very file.

Reply:
SOLVED - Added constants